sstables/storage.cc

```diff
@@ -36,10 +36,57 @@
 
 namespace sstables {
 
+class dir_wrapper final {
+    std::filesystem::path _pathname;
+    file _file;
+    bool _file_stale;
+
+public:
+    explicit dir_wrapper(std::filesystem::path pathname) : _pathname(std::move(pathname)) {};
+    dir_wrapper(const dir_wrapper&) = delete;
+    dir_wrapper& operator=(const dir_wrapper&) = delete;
+    dir_wrapper(dir_wrapper&&) = delete;
+    dir_wrapper& operator=(dir_wrapper&&) = delete;
```

Comment:
not sure why the move ctor and assignment operator need to be deleted

Reply:
Multiple instances of `dir_wrapper` aren't supposed to exist, therefore I'd minimally delete the copy constructor. And then the "rule of five" says, if we touch any one of those five member functions (destructor, copy constructor, move constructor, copy assignment operator, and move assignment operator), then be explicit about all of them.

Reply:
OK, in v5, I'll explicitly default the move ctor and the move assignment operator. AIUI, these perform memberwise movement. Each of the members
~~~
    std::filesystem::path _pathname;
    file _file;
~~~
support move semantics; the member `bool _file_stale` is supposed to go away in v5, and the new member `_close_fut` (of type `future`) also supports move semantics, AFAICT.

Reply:
These changes are listed in the v5 changelog; resolving this thread.